plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergMaterializedViewTest.java

```diff
@@ -1109,6 +1112,94 @@ public void testRefreshWithCompaction()
         assertUpdate("DROP TABLE %s".formatted(sourceTableName));
     }
 
+    @Test
+    void testPreviousSnapshotCleanupDuringRefresh()
+            throws IOException
+    {
+        String sourceTableName = "source_table" + randomNameSuffix();
+        String materializedViewName = "test_materialized_view" + randomNameSuffix();
+
+        // create source table and an MV
+        assertUpdate("CREATE TABLE " + sourceTableName + " (a int, b varchar)");
+        assertUpdate("INSERT INTO " + sourceTableName + " VALUES (1, 'abc'), (2, 'def')", 2);
+        assertUpdate("CREATE MATERIALIZED VIEW " + materializedViewName + " AS SELECT a, b FROM " + sourceTableName + " WHERE a < 3 OR a > 5");
+        // Until first MV refresh no data files are created hence perform first MV refresh to get data files created for the MV
+        assertUpdate("REFRESH MATERIALIZED VIEW " + materializedViewName, 2);
+
+        TrinoFileSystem fileSystemFactory = getFileSystemFactory(getQueryRunner()).create(ConnectorIdentity.ofUser("test"));
+
+        // Identify different types of files containing in an MV
+        Location metadataLocation = Location.of(getStorageMetadataLocation(materializedViewName));
+        FileIterator tableFiles = fileSystemFactory.listFiles(Location.of(metadataLocation.toString().substring(0, metadataLocation.toString().indexOf("/metadata"))));
+        ImmutableSet.Builder<FileEntry> dataFiles = ImmutableSet.builder();
```

Comment:
you can simply replace dataFiles with previousDataFiles, latestDataFiles with currentDataFiles etc.  Additionally use  assertThat(currentMetadataFiles.build()).containsAll(previousMetadataFiles.build()) 
      assertThat(currentManifestsFiles.build()).doesNotContainAnyElementsOf(previousManifestsFiles.build());